app/src/lib/stats/stats-store.ts

```diff
@@ -355,6 +357,18 @@ export class StatsStore {
     }))
   }
 
+  public recordRepoClicked(repoHasIndicator: boolean): Promise<void> {
```

Comment:
Introduced a variable to encompass that expression as per [your suggestion here](https://github.com/desktop/desktop/pull/4770#discussion_r197866420), so this shouldn't be an issue anymore :)

Reply:
Alternatively, `recordRepoClicked({ indicator: true })`

Reply:
@vanessayuenn not sure I understand. Was this comment meant for another line? This line was about introducing an enum to encode what the argument means.

Reply:
@iAmWillShepherd Using an enum in place of a boolean seems like an overkill to me, and the `hasXYZ` naming convention should make it self explanatory that either true/false is expected as an argument. Sorry I wasn't clear the first time around.

Reply:
You're right, enum's may be overkill, but I figured it was an easier solution that the alternative which is to have 2 functions. Flag paremeters in functions is actually a code smell (here are some articles on the topic:  [article 1](https://www.martinfowler.com/bliki/FlagArgument.html) [article 2](https://8thlight.com/blog/dariusz-pasciak/2015/05/28/alternatives-to-boolean-parameters.html))

We actually have examples of this all throughout the code base, but now that we will get to focus on paying down tech debt and improving quality, may as well start now 👍

Reply:
Let's all keep in mind that the very next method here does the exact same sin:

```ts
public async setOptOut(optOut: boolean): Promise<void> {
```

I'm fine with leaving this as-is - if we find ourselves looking for more complex analytics then this will need to be evolved anyway.